pkg/registry/registry.go

```diff
@@ -0,0 +1,211 @@
+package registry
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/suse-edge/edge-image-builder/pkg/fileio"
+	"github.com/suse-edge/edge-image-builder/pkg/http"
+	"github.com/suse-edge/edge-image-builder/pkg/image"
+	"go.uber.org/zap"
+)
+
+type helmClient interface {
```

Comment:
There's no hard rule as to where it should be defined - I did it here since it's only used in `registry.go` in the first place. I tend to keep as many type definitions as possible in the same file, especially when one is embedding others. It's a personal preference, though.